src/libpfc.c

```diff
@@ -382,7 +382,28 @@ static const struct EVENT EVENT_LIST[256] = {
     {0x00, NULL             , NULL}
 };
 
+/* Error Codes */
+#define PFC_ERR_OPENING_SYSFILE (-10)
+#define PFC_ERR_PWRITE_FAILED   (-11) // a pwrite() call returned error (check errno?)
+#define PFC_ERR_PWRITE_TOO_FEW  (-12) // a pwrite() call wrote less than the expected number of bytes
+#define PFC_ERR_CPU_PIN_FAILED  (-13) // cpu pinning failed, probably in sched_setaffinity
+#define PFC_ERR_CR4_PCE_NOT_SET (-14) // drive reported that cr4.pce wasn't set, or there was somehow an issue reading it
+#define PFC_ERR_AFFINITY_FAILED (-15) // setting CPU affinity failed (perhaps affinity is set externally excluding CPU 0?)
+#define PFC_ERR_READING_MASKS   (-16) // didn't read the expected number of mask bytes from the sysfs
 
+typedef struct {
+    int errorCode;
+    const char *message;
+} ErrorInfo;
+
+ErrorInfo AllErrors[] = {
```

Comment:
The table we use for lookup of the error strings. So to add a new error you just add a `#define` above and a line in this table.